paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/lookup/FixedBucketFromPkExtractor.java

```diff
@@ -62,49 +49,16 @@ public FixedBucketFromPkExtractor(TableSchema schema) {
                         schema.trimmedPrimaryKeys().stream()
                                 .mapToInt(schema.primaryKeys()::indexOf)
                                 .toArray());
-        this.partitionProjection =
-                CodeGenUtils.newProjection(
-                        schema.logicalPrimaryKeysType(),
-                        schema.partitionKeys().stream()
-                                .mapToInt(schema.primaryKeys()::indexOf)
-                                .toArray());
-        this.logPrimaryKeyProjection =
-                CodeGenUtils.newProjection(
-                        schema.logicalRowType(), schema.projection(schema.primaryKeys()));
     }
 
-    @Override
-    public void setRecord(InternalRow record) {
-        this.primaryKey = record;
-    }
-
-    @Override
-    public BinaryRow partition() {
-        return partitionProjection.apply(primaryKey);
-    }
-
-    private BinaryRow bucketKey() {
+    public int bucket(int numBuckets, InternalRow primaryKey) {
+        BinaryRow bucketKey;
         if (sameBucketKeyAndTrimmedPrimaryKey) {
-            return trimmedPrimaryKey();
+            bucketKey = trimmedPrimaryKeyProjection.apply(primaryKey);
```

Comment:
No need for this check. Unlike `KeyAndBucketExtractor`, trimmed primary key is not needed in this class.

Actually, why do you need two classes to implement this? Just put all these code into `PrimaryKeyPartialLookupTable`.